hibernate-core/src/main/java/org/hibernate/type/CharacterArrayClobType.java

```diff
@@ -33,8 +39,23 @@ public String getName() {
 	}
 
 	@Override
-	public <X> BasicType<X> resolveIndicatedType(JdbcTypeDescriptorIndicators indicators) {
-		//noinspection unchecked
+	public <X> BasicType<X> resolveIndicatedType(
+			JdbcTypeDescriptorIndicators indicators,
+			JavaTypeDescriptor<X> domainJtd) {
+		if ( domainJtd != null && domainJtd.getJavaTypeClass() == char[].class ) {
+			// domainJtd is a `char[]` instead of a `Character[]`....
+			final TypeConfiguration typeConfiguration = indicators.getTypeConfiguration();
+			final JdbcTypeDescriptorRegistry jdbcTypeRegistry = typeConfiguration.getJdbcTypeDescriptorRegistry();
+			final JdbcTypeDescriptor jdbcType = indicators.isNationalized()
+					? jdbcTypeRegistry.getDescriptor( Types.NCLOB )
+					: jdbcTypeRegistry.getDescriptor( Types.CLOB );
+
+			return typeConfiguration.getBasicTypeRegistry().resolve(
+					typeConfiguration.getJavaTypeDescriptorRegistry().getDescriptor( char[].class ),
```

Comment:
Shouln't this use `Character[]`?

```suggestion
					typeConfiguration.getJavaTypeDescriptorRegistry().getDescriptor( Character[].class ),
```

Reply:
Or maybe just pass the `domainJtd`?

Reply:
Yes, it should be the `domainJtd`.  Cases like this are why I added `domainJtd` to the args in the first place.  Just missed changing (at least) this one